### DISCUSSION TITLE
Basic technical question regarding failed delivery

### DISCUSSION THREAD

**Discussion by PhillipAnd** [mastodon/mastodon#22435] [Q&A]
With the recent influx of new users, many Mastodon instances are experiencing intermittent periods of being offline. Therefore I think the following question is important:

Say Bob on Instance B follows Alice on Instance A. If Alice creates a toot, Instance A will attempt to post to Instance B's inbox. What happens if Instance B gives an error or is non-responsive at that time? Does it get recycled in the push queue for another attempt later? Or does Instance A completely give up at that point (meaning it will never appear in Instance B's federated timeline or Bob's personal timeline)?

**Comment by nightpool** [comment]
retries are handled by the sidekiq retry behavior defined for that worker,
or by the default sidekiq behavior (I think 20 retries with exponential
back off?) if no specific retry behavior is defined.
